### PR TITLE
Support tracking more apps beyond Xcode

### LINKED ISSUE #15: Support tracking more apps beyond Xcode
* default on for all apps
* ability to toggle tracking off per app in settings menu
* start with Figma Desktop or Postman just to get it working for one app, can add more later

### PR DESCRIPTION
**Pull request by MMavrisPaleBlue**
Screenshot 2023-05-10 at 10 28 31 AM

Resolves #15.

### CODE REVIEW

File: WakaTime/Helpers/AppInfo.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//
+//  AppInfo.swift
```
Comment (alanhamlett) [inline WakaTime/Helpers/AppInfo.swift:2]: Is there a way to prevent Xcode adding these headers?

**Review by alanhamlett — APPROVED** [review]
The UI looks great! However, I did notice Postman isn't generating any heartbeats so we probably need changes in the file watcher since it's document structure is different from Xcode.

File: WakaTime/Watcher.swift [hunk 2]
```diff
@@ -46,7 +44,7 @@ class Watcher: NSObject {
     private func handleAppChanged(_ app: NSRunningApplication) {
         if app != activeApp {
             NSLog("App changed from \(activeApp?.localizedName ?? "nil") to \(app.localizedName ?? "nil")")
-            if let newAppID = app.bundleIdentifier, Watcher.appIDsToWatch.contains(newAppID) {
+            if let newAppID = app.bundleIdentifier, MonitoringManager.isAppMonitored(for: newAppID) {
```
Comment (alanhamlett) [inline WakaTime/Watcher.swift:47]: We should change `setXcodeVersion(app)` below to update a hash of versions for bundle ids so we don't accidentally set Xcode's version to Postman's version. Then when sending the heartbeat and building the `User-Agent` string, we use the corresponding version for the heartbeat's bundle id. We should also change the name of the app in that string to match the app we're tracking.

**Comment by MMavrisPaleBlue** [comment]
> Postman isn't sending heartbeats, probably because of one or many of these reasons:
> 
> * it could be triggering different AX events than Xcode: https://github.com/wakatime/macos-wakatime/pull/67/files#diff-8353e6f47a573aaa9401fba92dbd4587167525c0ed4852884d542f01cc065a80R165
> * Maybe getting the currently focused document path needs to be different for Postman https://github.com/wakatime/macos-wakatime/pull/67/files#diff-8353e6f47a573aaa9401fba92dbd4587167525c0ed4852884d542f01cc065a80R78

I will take a look, thanks

**Comment by alanhamlett** [comment]
After observing all AXNotification events, Postman doesn't trigger any of them:

```swift
            let notifications = [
                kAXAnnouncementKey,
                kAXAnnouncementRequestedNotification,
                kAXApplicationActivatedNotification,
                kAXApplicationDeactivatedNotification,
                kAXApplicationHiddenNotification,
                kAXApplicationShownNotification,
                kAXCreatedNotification,
                kAXDrawerCreatedNotification,
                kAXHelpTagCreatedNotification,
                kAXLayoutChangedNotification,
                kAXMainWindowChangedNotification,
                kAXMenuClosedNotification,
                kAXMenuItemSelectedNotification,
                kAXMenuOpenedNotification,
                kAXMovedNotification,
                kAXPriorityKey,
                kAXResizedNotification,
                kAXRowCollapsedNotification,
                kAXRowCountChangedNotification,
                kAXRowExpandedNotification,
                kAXSelectedCellsChangedNotification,
                kAXSelectedChildrenChangedNotification,
                kAXSelectedChildrenMovedNotification,
                kAXSelectedColumnsChangedNotification,
                kAXSelectedRowsChangedNotification,
                kAXSheetCreatedNotification,
                kAXTitleChangedNotification,
                kAXUIElementDestroyedNotification,
                kAXUIElementsKey,
                kAXUnitsChangedNotification,
                kAXValueChangedNotification,
                kAXWindowCreatedNotification,
                kAXWindowDeminiaturizedNotification,
                kAXWindowMiniaturizedNotification,
                kAXWindowMovedNotification,
                kAXWindowResizedNotification,
            ]
            for n in notifications {
                try observer.add(notification: n, element: element, refcon: this)
            }
```

File: WakaTime/Helpers/AppInfo.swift (hunk 1 above)
Comment (MMavrisPaleBlue) [reply WakaTime/Helpers/AppInfo.swift:2]: I will remove them

**Review by alanhamlett — APPROVED** [review]

**Pull request merged**